src/components/OpenWith/OpenWith.js

```diff
@@ -112,7 +121,26 @@ class OpenWith extends PureComponent<Props, State> {
      * @return {void}
      */
     componentDidMount() {
-        const { fileId }: Props = this.props;
+        this.fetchOpenWithData();
+    }
+
+    /**
+     * Fetches file extension and Open With data.
+     *
+     * @return {void}
+     */
+    fetchOpenWithData() {
+        const { fileId, extension }: Props = this.props;
+        if (extension) {
+            this.setState({ extension });
+        } else {
+            this.api
+                .getFileAPI()
+                .getFile(fileId, this.fetchExtensionSuccessCallback, noop, {
```

Comment:
There's no dependency between this call (when there's no extension) with the `getOpenWithIntegrations` call right?

Reply:
Correct, the icon can async load before or after we get back integrations. We'll use a generic icon until this comes back.